roles/translator/src/status.rs

```diff
@@ -0,0 +1,46 @@
+use crate::error::Error;
+
+#[derive(Debug)]
+pub enum State<'a> {
+    Shutdown(Error<'a>),
+    Healthy(String),
+}
+
+#[derive(Debug)]
+pub struct Status<'a> {
+    pub state: State<'a>,
+}
+
+/// # Description
+/// This macro handles errors inserting error handling logic for a given `Result<T, crate::error::Error<'a>>`
+/// it is used by passing in a `Sender<crate::status::Status>` as the first parameter
+/// and a `Result<T, crate::error::Error<'a>>` as the second parameter.
+/// NOTE: can only be used within async functions since status needs to be send over async channel
+/// # Example
+/// ```
+/// let (tx_status: Sender<Status>, rx_status) = async_channel::unbounded();
+/// let variable = handle_result!(
+///     tx_status,
+///     type_.try_into()
+/// );
+/// ```
+#[macro_export]
+macro_rules! handle_result {
```

Comment:
Still have to deeper review it but this seem like something that should stay in `/utils/` so that we can have a consistent way to manage this kind of issues across all the SRI codebase.

Reply:
We should modify `network_helpers` so that we can use this macro when we are trying to read/write from the socket and in case of an error fail. So now that we have it I think that we can remove the panics from `network_helpers`